### PR TITLE
Reduce memory alloc in FindCols

### PR DESCRIPTION
**Pull request by innerr**
Signed-off-by: Liu Cong <liucong@pingcap.com>

### What problem does this PR solve? 
Reduce the memory allocating in function FindCols

### What is changed and how it works?
Simply switch to pre-allocating

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes
  - Simple change in the function FindCols and some gofmt result

Side effects

 - Possible performance regression

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: table/column.go [hunk 1]
```diff
@@ -88,16 +88,16 @@ func ToColumn(col *model.ColumnInfo) *Column {
...
+	for i, name := range names {
 		col := FindCol(cols, name)
 		if col != nil {
 			rcols = append(rcols, col)
 		} else if name == model.ExtraHandleName.L && !pkIsHandle {
 			col := &Column{}
 			col.ColumnInfo = model.NewExtraHandleColInfo()
 			col.ColumnInfo.Offset = len(cols)
-			rcols = append(rcols, col)
+			rcols[i] = col
```
Comment (siddontang) [inline table/column.go]: seem you still need to use append here.

Comment (innerr) [reply table/column.go]: Already fixed.
And the func FindCol also a slow liner-find

Comment (siddontang) [reply table/column.go]: yes, I have already told @zz-jason about this, maybe we can use a map to speed up it.

Comment (zz-jason) [reply table/column.go]: @winoros PTAL

Comment (innerr) [reply table/column.go]: @siddontang https://github.com/pingcap/tidb/pull/14903 using a map

**Pull request merged**